### PR TITLE
add geolocation samples

### PR DESCRIPTION
**Pull request by patrickkettner**

### CODE REVIEW

File: functional-samples/cookbook.geolocation-contentscript/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+This recipe shows how to get `geolocation` access within a [content script][5]
+
+## Context
+
+In extensions, geolocation uses the same web APIs you are used to. Instead, if you want to get geolocation information, use the same [`navigator.geolocation`](geolocation) DOM API that any web site normally would. This demo shows how to access this data in a [content script][6]. We also have demos for a [Service Worker][4](via [offscreenDocument][2], and using a [popup][3]
+
+## Running this extension
+
+1. Clone this repository.
+1. Load this directory in Chrome as an [unpacked extension][2].
```
Comment (AmySteam) [inline functional-samples/cookbook.geolocation-contentscript/README.md:10]:
```suggestion
1. Load this directory in Chrome as an [unpacked extension][1].
```

File: functional-samples/cookbook.geolocation-contentscript/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+This recipe shows how to get `geolocation` access within a [content script][5]
+
+## Context
+
+In extensions, geolocation uses the same web APIs you are used to. Instead, if you want to get geolocation information, use the same [`navigator.geolocation`](geolocation) DOM API that any web site normally would. This demo shows how to access this data in a [content script][6]. We also have demos for a [Service Worker][4](via [offscreenDocument][2], and using a [popup][3]
```
Comment (AmySteam) [inline functional-samples/cookbook.geolocation-contentscript/README.md:5]:
```suggestion
To get geolocation information in extensions, use the same [`navigator.geolocation`][6] DOM API that any website normally would. This demo shows how to access this data in a [content script][5]. We also have demos for a [Service Worker][4](via [offscreenDocument][2], and using a [popup][3].
```

File: functional-samples/cookbook.geolocation-offscreen/README.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+This recipe shows how to get `geolocation` access within a service worker
+
+## Context
+
+In extensions, geolocation uses the same web APIs you are used to. Instead, if you want to get geolocation information, use the same [`navigator.geolocation`](geolocation) DOM API that any web site normally would. This demo shows how to access this data in a Service Worker (via [offscreenDocument][2]. We also have samples for a [popup][3], or a [content script][4]
```
Comment (AmySteam) [inline functional-samples/cookbook.geolocation-offscreen/README.md:5]:
```suggestion
To get geolocation information in extensions, use the same [`navigator.geolocation`][6] DOM API that any website normally would. This demo shows how to access this data in a Service Worker (via [offscreenDocument][2]. We also have samples for a [popup][3], or a [content script][4].
```

File: functional-samples/cookbook.geolocation-offscreen/README.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+## Running this extension
+
+1. Clone this repository.
+1. Load this directory in Chrome as an [unpacked extension][2].
```
Comment (AmySteam) [inline functional-samples/cookbook.geolocation-offscreen/README.md:10]:
```suggestion
1. Load this directory in Chrome as an [unpacked extension][1].
```

File: functional-samples/cookbook.geolocation-offscreen/service_worker.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+const OFFSCREEN_DOCUMENT_PATH = '/offscreen.html';
+
+chrome.runtime.onMessage.addListener(handleMessages);
+
+let creating; // A global promise to avoid concurrency issues
+
+async function hasDocument() {
+  // Check all windows controlled by the service worker to see if one
+  // of them is the offscreen document with the given path
+  const offscreenUrl = chrome.runtime.getURL(OFFSCREEN_DOCUMENT_PATH);
```
Comment (AmySteam) [inline functional-samples/cookbook.geolocation-offscreen/service_worker.js:24]: In service workers, you can access extension resources directly. the `runtime.getURL()` was created to access extension assets in content scripts.

File: functional-samples/cookbook.geolocation-contentscript/README.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+1. Clone this repository.
+1. Load this directory in Chrome as an [unpacked extension][2].
+1. Open the Extension menu and click the extension named "Geolocation - content script".
+
+[1]: https://developer.chrome.com/docs/extensions/mv3/getstarted/development-basics/#load-unpacked
+[2]: https://developer.chrome.com/docs/extensions/reference/offscreen/
+[3]: functional-samples/cookbook.geolocation-popup
+[4]: functional-samples/cookbook.geolocation-offscreen
+[5]: https://developer.chrome.com/docs/extensions/mv3/content_scripts/
```
Comment (AmySteam) [inline functional-samples/cookbook.geolocation-contentscript/README.md:17]:
```suggestion
[5]: https://developer.chrome.com/docs/extensions/mv3/content_scripts/
[6]: https://developer.mozilla.org/docs/Web/API/Navigator/geolocation
```

File: functional-samples/cookbook.geolocation-popup/README.md [hunk 7]
```diff
@@ -0,0 +1,17 @@
+This recipe shows how to get `geolocation` access within a popup
+
+## Context
+
+In extensions, geolocation uses the same web APIs you are used to. Instead, if you want to get geolocation information, use the same [`navigator.geolocation`](geolocation) DOM API that any web site normally would. This demo shows how to access this data in a [popup][3]. We also have demos for a [Service Worker][5](via [offscreenDocument][2], and using a [content script][4]
```
Comment (AmySteam) [inline functional-samples/cookbook.geolocation-popup/README.md:5]:
```suggestion
To get geolocation information in extensions, use the same [`navigator.geolocation`][6] DOM API that any website normally would.This demo shows how to access this data in a [popup][3]. We also have demos for a [Service Worker][5](via [offscreenDocument][2], and using a [content script][4]
```

File: functional-samples/cookbook.geolocation-offscreen/README.md [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+1. Open the Extension menu and click the extension named "Geolocation - offscreen".
+
+[1]: https://developer.chrome.com/docs/extensions/mv3/getstarted/development-basics/#load-unpacked
+[2]: https://developer.chrome.com/docs/extensions/reference/offscreen/
+[3]: functional-samples/cookbook.geolocation-popup
+[4]: functional-samples/cookbook.geolocation-contentscript
```
Comment (AmySteam) [inline functional-samples/cookbook.geolocation-offscreen/README.md:16]:
```suggestion
[4]: functional-samples/cookbook.geolocation-contentscript
[6]: https://developer.mozilla.org/docs/Web/API/Navigator/geolocation
```

File: functional-samples/cookbook.geolocation-popup/README.md [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+
+1. Clone this repository.
+1. Load this directory in Chrome as an [unpacked extension][2].
+1. Open the Extension menu and click the extension named "Geolocation - popup".
+
+[1]: https://developer.chrome.com/docs/extensions/mv3/getstarted/development-basics/#load-unpacked
+[2]: https://developer.chrome.com/docs/extensions/reference/offscreen/
+[3]: functional-samples/cookbook.geolocation-popup
+[4]: functional-samples/cookbook.geolocation-contentscript
+[5]: functional-samples/cookbook.geolocation-offscreen
```
Comment (AmySteam) [inline functional-samples/cookbook.geolocation-popup/README.md:17]:
```suggestion
[5]: functional-samples/cookbook.geolocation-offscreen
[6]: https://developer.mozilla.org/docs/Web/API/Navigator/geolocation
```

**Review by AmySteam — CHANGES_REQUESTED** [review]
Thanks @patrickkettner!

Just left a few suggestions :)

File: functional-samples/cookbook.geolocation-offscreen/service_worker.js (hunk 5 above)
Comment (patrickkettner) [reply functional-samples/cookbook.geolocation-offscreen/service_worker.js:24]: removed!

File: functional-samples/cookbook.geolocation-offscreen/service_worker.js [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const OFFSCREEN_DOCUMENT_PATH = '/offscreen.html';
+
+chrome.runtime.onMessage.addListener(handleMessages);
```
Comment (sebastianbenz) [inline functional-samples/cookbook.geolocation-offscreen/service_worker.js:17]: It'd be nice to add a high-level function `getGeoLocation` here that returns the current position. I think it'd be good to also provide a snippet for the promise juggling that this requires.

**Review by sebastianbenz — COMMENTED** [review]
Great samples! One suggestion otherwise LGTM

**Review by AmySteam — CHANGES_REQUESTED** [review]
@patrickkettner so I decided to load and test the extensions by following the readme instructions. I've only done it with the content script sample so far. We need to include the extra steps to navigate to www.example.com and inspect the content script to see the location. 

Unless you were planning on displaying it as part of the H1 header, but that would need a bit more code.

File: functional-samples/cookbook.geolocation-offscreen/service_worker.js [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+  if (!(await hasDocument())) {
+    // create offscreen document
+    if (creating) {
+      await creating;
+    } else {
+      creating = chrome.offscreen.createDocument({
+        url: path,
+        reasons: [
+          chrome.offscreen.Reason.GEOLOCATION ||
+            chrome.offscreen.Reason.DOM_SCRAPING
```
Comment (cuylerstuwe) [inline functional-samples/cookbook.geolocation-offscreen/service_worker.js:40]: What's the purpose / effect of using `||` here?

Is there some sort of case where `chrome.offscreen.Reason.GEOLOCATION` might not be defined? 🤔

Comment (patrickkettner) [reply functional-samples/cookbook.geolocation-offscreen/service_worker.js:40]: Hi @cuylerstuwe! Your hunch was correct. It is actually not in Chromium as of this moment. You can see the path [here](https://chromium-review.googlesource.com/c/chromium/src/+/4605111) though will be patched in ~tomorrow. Once added, it is the "correct" choice, but `DOM_SCRAPING` is the least incorrect choice for production code after that.

Comment (cuylerstuwe) [reply functional-samples/cookbook.geolocation-offscreen/service_worker.js:40]: It's interesting to see a Google-backed example of _"using the least-incorrect option"_.

I've wondered about this for a variety of use-cases, e.g., whether WebSockets would be allowed despite not having a reason specifically provided for them.

This seems to skew more in the favor of maybe using e.g. `DOM_SCRAPING` for WebSockets.

Comment (patrickkettner) [reply functional-samples/cookbook.geolocation-offscreen/service_worker.js:40]: Specifically in regard to reasons, I like to add cases as we find them. For the general case of "websockets", `DOM_SCRAPING` _may_ be the best option as of now. It really depends on what you are doing. I agree there needs to be a general websocket/real-time-communication reason added, however

**Comment by patrickkettner** [comment]
@AmySteam great call! I pushed up dc744fa to document that better.

File: functional-samples/cookbook.geolocation-offscreen/service_worker.js (hunk 10 above)
Comment (patrickkettner) [reply functional-samples/cookbook.geolocation-offscreen/service_worker.js:17]: Are you suggesting removing the current general `handleMessages` approach, and replace it with a `getGeolocation`, or supplementing the existing code with a convenience `getGeolocation`?

For the later, its a bit awkward, but the simplest addition to the existing code would look like

```js
function getGeolocation() {
  return new Promise(async (resolve, reject) => {
    
    chrome.runtime.onMessage.addListener(msg => {
      if (msg.type === 'geolocation-success') {
        resolve(msg.data);
      } else if (msg.type === 'geolocation-err') {
        reject(`unable to get geolocation data - ${msg.data}`)
      }
    })
    
    await sendMessageToOffscreenDocument('get-geolocation')
  })
}
```


Happy to do the former, just wanted to check first

Comment (sebastianbenz) [reply functional-samples/cookbook.geolocation-offscreen/service_worker.js:17]: Afaik the later is not recommended as it nests callback registration in a method call. The former would be better imo.

File: functional-samples/cookbook.geolocation-offscreen/README.md [hunk 12]
```diff
@@ -0,0 +1,17 @@
+This recipe shows how to get `geolocation` access within a service worker
+
+## Context
+
+To get geolocation information in extensions, use the same [`navigator.geolocation`][6] DOM API that any website normally would. This demo shows how to access this data in a Service Worker (via [offscreenDocument][2]. We also have samples for a [popup][3], or a [content script][4].
+
+## Running this extension
```
Comment (AmySteam) [inline functional-samples/cookbook.geolocation-offscreen/README.md:7]: I tested the offscreen docs sample. When I click on the action icon when I'm on the chrome://extensions page, nothing happens. I inspected the SW and I didn't see any logs 

Next, I tried navigating to example.com and clicking on the action icon. I inspected the service worker, but there was no log.

We should probably add more steps.

File: functional-samples/cookbook.geolocation-offscreen/service_worker.js (hunk 10 above)
Comment (patrickkettner) [reply functional-samples/cookbook.geolocation-offscreen/service_worker.js:17]: yeah, I don't think there is a way around nesting, as you need to have the reference to the `resolve` of the initiator. I'll refactor

**Comment by patrickkettner** [comment]
Updated the code based on our conversation earlier today. There isn't an obvious place to show the response. @AmySteam @sebastianbenz is the console an acceptable place for people to see the response for the offscreenDocument version?

**Review by sebastianbenz — APPROVED** [review]
The offscreen doc solution looks great now! sendResponse makes this so much easier.

**Comment by patrickkettner** [comment]
@sebastianbenz for sure! @oliverdunk was correct right about using it. I incorrectly thought it wasn't usable in that situation because the message handler on the offscreenDocument side of things wasn't returning the expected value in my testing. This was because I still had it set as `async`. In order to send an async value from the offscreenDoc, the message handler needed to be explicitly returning `true`, not just something truthy. Since it was async, it returned a promise. 

Did you get a chance to read my [last question ](https://github.com/GoogleChrome/chrome-extensions-samples/pull/944#issuecomment-1592188344) by chance?

**Comment by sebastianbenz** [comment]
Yes, console output sgtm. All good from my side what about you Amy?

**Comment by AmySteam** [comment]
> Updated the code based on our conversation earlier today. There isn't an obvious place to show the response. @AmySteam @sebastianbenz is the console an acceptable place for people to see the response for the offscreenDocument version?

**IDEA** 💡: It might be fun to add the geolocation to the [action title (tooltip)](https://developer.chrome.com/docs/extensions/reference/action/#tooltip-title), using [action.setTitle()](https://developer.chrome.com/docs/extensions/reference/action/#method-setTitle). 

Developers can see it by hovering on the extension action. For example
image

**Comment by patrickkettner** [comment]
@AmySteam love it! will PR shortly

**Review by AmySteam — APPROVED** [review]
Okay, this I like 👍 

@patrickkettner Thanks for taking on this challenge! I really like how the offscreen sample turned out 💯

File: functional-samples/cookbook.geolocation-contentscript/README.md [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+## Running this extension
+
+1. Clone this repository.
+1. Load this directory in Chrome as an [unpacked extension][1].
+1. Navigate to example.com
+1. Inspect the page to see the location being logged to the console.
+
+[1]: https://developer.chrome.com/docs/extensions/mv3/getstarted/development-basics/#load-unpacked
+[2]: https://developer.chrome.com/docs/extensions/reference/offscreen/
+[3]: functional-samples/cookbook.geolocation-popup
```
Comment (oliverdunk) [inline functional-samples/cookbook.geolocation-contentscript/README.md:16]: I'm not sure that these relative links will work unless we add a / to make them relative to the root?

**Review by oliverdunk — CHANGES_REQUESTED** [review]
Left a few comments but this is so great - thanks for all of the time you've spent on it! Especially love the icons.

File: functional-samples/cookbook.geolocation-popup/popup.js [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+
+  return `${deg}&deg;${min}'${sec}"${direction}`;
+};
+
+navigator.geolocation.getCurrentPosition(
+  (loc) => {
+    const { coords } = loc;
+    const { latitude, longitude } = coords;
+
+    header.innerHTML = `position: ${C(latitude, true)}, ${C(longitude)}`;
```
[the remark this answers is not in the document]
Comment (patrickkettner) [reply functional-samples/cookbook.geolocation-popup/popup.js:36]: ah, great catch. I copied from the other section.

File: functional-samples/cookbook.geolocation-offscreen/service_worker.js [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+    chrome.action.setIcon({ path: 'images/lightgreen.png' });
+    await setTitle('locating...');
+
+    locating = await getGeolocation();
+    const { coords } = locating;
+    const { latitude, longitude } = coords;
+
+    chrome.action.setIcon({ path: 'images/green.png' });
+    await setTitle(`position: ${C(latitude, true)}, ${C(longitude)}`);
+    locating = null;
```
[the remark this answers is not in the document]
Comment (patrickkettner) [reply functional-samples/cookbook.geolocation-offscreen/service_worker.js:104]: great call

File: functional-samples/cookbook.geolocation-contentscript/content-script.js [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Registering this listener when the script is first executed ensures that the
+// offscreen document will be able to receive messages when the promise returned
+// by `offscreen.createDocument()` resolves.
+
+const header = document.querySelector('h1');
+
+const C = (coords, isLat) => {
```
[the remark this answers is not in the document]
Comment (patrickkettner) [reply functional-samples/cookbook.geolocation-contentscript/content-script.js:21]: to be terse on [this line](https://github.com/GoogleChrome/chrome-extensions-samples/pull/944/files/a2cb681618e509e791af4de0280375b0e989a2fc#diff-50b14a5fc3cde965f1e7c9fed17634736ea2008b2d3357ae435b62e2ae1cee1dR35). Happy to update it

File: functional-samples/cookbook.geolocation-offscreen/README.md [hunk 17]
```diff
@@ -0,0 +1,19 @@
+This recipe shows how to get `geolocation` access within a service worker
+
+## Context
+
+To get geolocation information in extensions, use the same [`navigator.geolocation`][6] DOM API that any website normally would. This demo shows how to access this data in a Service Worker (via [offscreenDocument][2]. We also have samples for a [popup][3], or a [content script][4].
```
[the remark this answers is not in the document]
Comment (patrickkettner) [reply functional-samples/cookbook.geolocation-offscreen/README.md:5]: sorry, which point is being dittoed?

File: functional-samples/cookbook.geolocation-popup/README.md [hunk 18]
```diff
@@ -0,0 +1,18 @@
+This recipe shows how to get `geolocation` access within a popup
+
+## Context
+
+To get geolocation information in extensions, use the same [`navigator.geolocation`][6] DOM API that any website normally would.This demo shows how to access this data in a [popup][3]. We also have demos for a [Service Worker][5](via [offscreenDocument][2], and using a [content script][4]
```
[the remark this answers is not in the document]
Comment (patrickkettner) [reply functional-samples/cookbook.geolocation-popup/README.md:5]: ditto to my question about what is being dittoed

File: functional-samples/cookbook.geolocation-offscreen/README.md (hunk 17 above)
[the remark this answers is not in the document]
Comment (oliverdunk) [reply functional-samples/cookbook.geolocation-offscreen/README.md:7]: Oops, sorry - just the formatting changes here: https://github.com/GoogleChrome/chrome-extensions-samples/pull/944#discussion_r1231148022

File: functional-samples/cookbook.geolocation-popup/README.md (hunk 18 above)
[the remark this answers is not in the document]
Comment (oliverdunk) [reply functional-samples/cookbook.geolocation-popup/README.md:7]: Haha, still this one: https://github.com/GoogleChrome/chrome-extensions-samples/pull/944#discussion_r1231148022

File: functional-samples/cookbook.geolocation-contentscript/README.md (hunk 13 above)
Comment (patrickkettner) [reply functional-samples/cookbook.geolocation-contentscript/README.md]: fixed!

File: functional-samples/cookbook.geolocation-popup/popup.js (hunk 14 above)
[the remark this answers is not in the document]
Comment (patrickkettner) [reply functional-samples/cookbook.geolocation-popup/popup.js]: updated to innerText

File: functional-samples/cookbook.geolocation-contentscript/manifest.json [hunk 19]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "Geolocation - content script",
+  "version": "1.0",
+  "manifest_version": 3,
+  "content_scripts": [
+    {
+      "matches": ["https://example.com/*"],
+      "match_origin_as_fallback": true,
```
[the remark this answers is not in the document]
Comment (patrickkettner) [reply functional-samples/cookbook.geolocation-contentscript/manifest.json]: removed

File: functional-samples/cookbook.geolocation-offscreen/README.md (hunk 17 above)
[the remark this answers is not in the document]
Comment (patrickkettner) [reply functional-samples/cookbook.geolocation-offscreen/README.md:7]: happy to update - is it "offscreen document", and not "Offscreen Document"?

File: functional-samples/cookbook.geolocation-offscreen/images/green.png
[the remark this answers is not in the document]
Comment (patrickkettner) [reply functional-samples/cookbook.geolocation-offscreen/images/green.png:1]: :D

File: functional-samples/cookbook.geolocation-contentscript/README.md [hunk 20]
```diff
@@ -0,0 +1,19 @@
+This recipe shows how to get `geolocation` access within a [content script][5]
+
+## Context
```
[the remark this answers is not in the document]
Comment (patrickkettner) [reply functional-samples/cookbook.geolocation-contentscript/README.md]: I add a title and change context to overview - any other places you would like to see more alignment?

File: functional-samples/cookbook.geolocation-popup/README.md (hunk 18 above)
[the remark this answers is not in the document]
Comment (patrickkettner) [reply functional-samples/cookbook.geolocation-popup/README.md:7]: :D for the sake of redundancy...https://github.com/GoogleChrome/chrome-extensions-samples/pull/944#discussion_r1231245539

File: functional-samples/cookbook.geolocation-contentscript/README.md [hunk 21]
```diff
@@ -0,0 +1,21 @@
+This recipe shows how to get `geolocation` access within a [content script][5]
```
Comment (oliverdunk) [inline functional-samples/cookbook.geolocation-contentscript/README.md:1]: Nit: Can we move this down between the header and "Overview"?

File: functional-samples/cookbook.geolocation-offscreen/README.md [hunk 22]
```diff
@@ -0,0 +1,21 @@
+This recipe shows how to get `geolocation` access within a service worker
```
Comment (oliverdunk) [inline functional-samples/cookbook.geolocation-offscreen/README.md:1]: Nit: Can we move this down between the header and "Overview"?

File: functional-samples/cookbook.geolocation-offscreen/README.md [hunk 23]
```diff
@@ -0,0 +1,21 @@
...
+
+# geolocation via offscreen document
+
+## Overview
+
+To get geolocation information in extensions, use the same [`navigator.geolocation`][6] DOM API that any website normally would. This demo shows how to access this data in a Service Worker (via [offscreen document][2]. We also have samples for a [popup][3], or a [content script][4].
```
Comment (oliverdunk) [inline functional-samples/cookbook.geolocation-offscreen/README.md:7]:
```suggestion
To get geolocation information in extensions, use the same [`navigator.geolocation`][6] DOM API that any website normally would. This demo shows how to access this data in a Service Worker (via [offscreen document][2]). We also have samples for a [popup][3], or a [content script][4].
```

File: functional-samples/cookbook.geolocation-popup/README.md [hunk 24]
```diff
@@ -0,0 +1,20 @@
+This recipe shows how to get `geolocation` access within a popup
```
Comment (oliverdunk) [inline functional-samples/cookbook.geolocation-popup/README.md:1]: Nit: Can we move this down between the header and "Overview"?

File: functional-samples/cookbook.geolocation-popup/README.md [hunk 25]
```diff
@@ -0,0 +1,20 @@
...
+
+# geolocation inside a popup
+
+## Overview
+
+To get geolocation information in extensions, use the same [`navigator.geolocation`][6] DOM API that any website normally would.This demo shows how to access this data in a [popup][3]. We also have demos for a [Service Worker][5](via [offscreen document][2], and using a [content script][4]
```
Comment (oliverdunk) [inline functional-samples/cookbook.geolocation-popup/README.md:7]:
```suggestion
To get geolocation information in extensions, use the same [`navigator.geolocation`][6] DOM API that any website normally would. This demo shows how to access this data in a [popup][3]. We also have demos for a [Service Worker][5](via [offscreen document][2]), and using a [content script][4]
```

**Review by oliverdunk — APPROVED** [review]
LGTM with nits. Thanks for all of the work on this!

File: functional-samples/cookbook.geolocation-popup/README.md (hunk 25 above)
Comment (patrickkettner) [reply functional-samples/cookbook.geolocation-popup/README.md]: fixed with [1dbac70](https://github.com/GoogleChrome/chrome-extensions-samples/pull/944/commits/1dbac7034801f93237ddfa43f3a9a44f9f7552eb)

File: functional-samples/cookbook.geolocation-offscreen/README.md (hunk 23 above)
Comment (patrickkettner) [reply functional-samples/cookbook.geolocation-offscreen/README.md]: fixed with [1dbac70](https://github.com/GoogleChrome/chrome-extensions-samples/pull/944/commits/1dbac7034801f93237ddfa43f3a9a44f9f7552eb)

File: functional-samples/cookbook.geolocation-popup/README.md (hunk 24 above)
Comment (patrickkettner) [reply functional-samples/cookbook.geolocation-popup/README.md]: fixed with [1dbac70](https://github.com/GoogleChrome/chrome-extensions-samples/pull/944/commits/1dbac7034801f93237ddfa43f3a9a44f9f7552eb)

File: functional-samples/cookbook.geolocation-contentscript/README.md (hunk 21 above)
Comment (patrickkettner) [reply functional-samples/cookbook.geolocation-contentscript/README.md]: fixed with [1dbac70](https://github.com/GoogleChrome/chrome-extensions-samples/pull/944/commits/1dbac7034801f93237ddfa43f3a9a44f9f7552eb)

File: functional-samples/cookbook.geolocation-offscreen/README.md (hunk 22 above)
Comment (patrickkettner) [reply functional-samples/cookbook.geolocation-offscreen/README.md]: fixed with [1dbac70](https://github.com/GoogleChrome/chrome-extensions-samples/pull/944/commits/1dbac7034801f93237ddfa43f3a9a44f9f7552eb)

**Pull request merged**